pdlua.c

```diff
@@ -2664,33 +2662,43 @@ static int pdlua_canvas_realizedollar(lua_State *L)
 
 static int pdlua_signal_setmultiout(lua_State *L)
 {
-    if (lua_islightuserdata(L, 1) && lua_isnumber(L, 2) && lua_isnumber(L, 3))
+    char msg[MAXPDSTRING];
+
+    if (!(lua_islightuserdata(L, 1) && lua_isnumber(L, 2) && lua_isnumber(L, 3)))
     {
-        t_pdlua *x = (t_pdlua *)lua_touserdata(L, 1);
-        PDLUA_DEBUG("pdlua_signal_setmultiout: x = %p", x);
-        int outidx = lua_tointeger(L, 2) - 1;
-        int nchans = lua_tointeger(L, 3);
-        if (nchans < 1) {
-            pd_error(NULL, "lua: invalid channel count: %d, setting to 1", nchans);
-            nchans = 1;  // Ensure at least one channel
-        }
-        PDLUA_DEBUG2("pdlua_signal_setmultiout: outidx = %d, nchans = %d", outidx, nchans);
-        
-        if (x && outidx >= 0 && outidx < x->sigoutlets)
-        {
+        pd_error(NULL, "%s: signal_setmultiout: invalid arguments", src_info(L, msg));
+        return 0;
+    }
+
+    t_pdlua *x = (t_pdlua *)lua_touserdata(L, 1);
+    int outidx = lua_tointeger(L, 2) - 1;
+    int nchans = lua_tointeger(L, 3);
+    if (!(x && outidx >= 0 && outidx < x->sigoutlets))
+    {
+        pd_error(NULL, "%s: signal_setmultiout: must be called from dsp method", src_info(L, msg));
```

Comment:
not sure if it's valid to assume that this error is due to the wrong context.

Reply:
i split these now in the following commit. still not sure about the assumption of the wrong context if x is not defined.

Reply:
the error messages might still be improved - but they already give a good hint on what might go wrong. closing this comment for now.